### PR TITLE
ui test branch

### PR DESCRIPTION
**Pull request by zajonesck**
new readme and tests refactored

### CODE REVIEW

**Comment by zajonesck** [comment]
check it out green

File: ui/__tests__/clientWorkout.test.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  describe("Sorting functionality", () => {
+    test.each([
+      ["workout_name", true, "Arm Day"],
+      ["workout_name", false, "Chest Day"],
+      ["difficulty", true, "Beginner"],
+      ["difficulty", false, "Intermediate"],
+      ["date", true, "2023-09-22"],
+      ["date", false, "2023-09-23"],
+    ])("should sort by %s in %s order", (column, ascending, expected) => {
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:34]: change `ascending` to `sortAscending`

File: ui/__tests__/clientWorkout.test.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  describe("Search functionality", () => {
+    test.each([
+      ["Chest Day", "workout_name", 1, "Chest Day"],
+      ["Calisthenics", "workout_name", 0],
+      ["cHeSt DaY", "workout_name", 1, "Chest Day"],
+      ["Intermediate", "difficulty", 1, "Intermediate"],
+      ["Intense", "notes", 1, "Intense"],
+    ])(
+      "should filter workouts by %s in %s column",
+      (query, column, expectedLength, expectedValue) => {
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:53]: change `expectedLength` to `expectedResults`

File: ui/__tests__/clientWorkout.test.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      });
+      expect(wrapper.vm.sortedWorkouts[0][column]).toBe(expected);
+    });
+  });
+
+  describe("Search functionality", () => {
+    test.each([
+      ["Chest Day", "workout_name", 1, "Chest Day"],
+      ["Calisthenics", "workout_name", 0],
+      ["cHeSt DaY", "workout_name", 1, "Chest Day"],
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:48]: test.each doesn't capture everything that's different about these tests. I recommend splitting these back up.
For example,  `["Chest Day", "workout_name", 1, "Chest Day"],` is testing a basic example of searching by name, `["Calisthenics", "workout_name", 0],` is testing that no results are returned, `["cHeSt DaY", "workout_name", 1, "Chest Day"],` is testing that the results are not case sensitive,

File: ui/__tests__/clientWorkout.test.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      date: "2023-09-22",
+      notes: "Light",
+    },
+  ];
+
+  beforeEach(() => {
+    wrapper = mount(ClientWorkout);
+  });
+
+  describe("Sorting functionality", () => {
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:26]: add a description about all the test cases

File: ui/__tests__/clientWorkout.test.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        date: "2023-09-21",
+        notes: "Very Intense",
+      },
+    ];
+
+    test("should display correct number of workouts per page", () => {
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        currentPageAssignedWorkouts: 1,
+        itemsPerPageAssignedWorkouts: 2,
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:94]: change to `itemsPerPage` - needs change in both test and component

File: ui/__tests__/clientWorkout.test.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        difficulty: "Advanced",
+        date: "2023-09-21",
+        notes: "Very Intense",
+      },
+    ];
+
+    test("should display correct number of workouts per page", () => {
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        currentPageAssignedWorkouts: 1,
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:93]: change to `currentPage` - needs change in both test and component

File: ui/__tests__/clientWorkout.test.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+      wrapper.setData({
+        clientWorkouts: clientWorkoutsData,
+        currentPageAssignedWorkouts: 2,
+        itemsPerPageAssignedWorkouts: 2,
+      });
+      expect(wrapper.vm.paginatedAssignedWorkouts.length).toBe(1); // As we only have 3 workouts in the sample data
+      expect(wrapper.vm.paginatedAssignedWorkouts[0].workout_name).toBe(
+        "Leg Day"
+      );
+    });
```
Comment (chelseagreen) [inline ui/__tests__/clientWorkout.test.js:125]: what happens if i have multiple pages and i sort by a column? Can you write a test for this?

**Review by chelseagreen — CHANGES_REQUESTED** [review]

File: ui/__tests__/clients.test.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = mount(Clients);
+  });
+
+  describe("Sorting functionality", () => {
+    const clientData = [
+      { first_name: "Zachary", last_name: "Jones", birth_day: "1995-05-15" },
+      { first_name: "Alice", last_name: "Smith", birth_day: "1990-03-10" },
```
Comment (chelseagreen) [inline ui/__tests__/clients.test.js:14]: Let's add a third item here

File: ui/__tests__/clients.test.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+      });
+      expect(wrapper.vm.sortedClients[0].first_name).toBe("Zachary");
+    });
+  });
+
+  describe("Search functionality", () => {
+    const clientData = [
+      { first_name: "Zachary", last_name: "Jones" },
+      { first_name: "Alice", last_name: "Smith" },
+    ];
```
Comment (chelseagreen) [inline ui/__tests__/clients.test.js:58]: Why is client data a different object here? You could use the same object in both describes

File: ui/__tests__/clients.test.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        last_name: "Jones",
+        birth_day: "1995-05-15",
+      },
+      {
+        id: 2,
+        first_name: "Alice",
+        last_name: "Smith",
+        birth_day: "1990-03-10",
+      },
+      { id: 3, first_name: "Bob", last_name: "Doe", birth_day: "1985-01-01" },
```
Comment (chelseagreen) [inline ui/__tests__/clients.test.js:92]: same thing here. clientData can be all one object decleared in the file and then used in all the describes

File: ui/__tests__/clients.test.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+      expect(wrapper.vm.totalPages).toBe(Math.ceil(clientData.length / 2));
+    });
+
+    test("should navigate to next page correctly", () => {
+      wrapper.setData({
+        clients: clientData,
+        currentPage: 2,
+        itemsPerPage: 2,
+      });
+      expect(wrapper.vm.paginatedClients.length).toBe(1); // As we only have 3 clients in the sample data
```
Comment (chelseagreen) [inline ui/__tests__/clients.test.js:120]: remove this dumb comment

File: ui/__tests__/exercise.test.js [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+              primary_body_part: "Triceps",
+              secondary_body_part: "Calves",
+            },
+            {
+              name: "Arm Day",
+              primary_body_part: "Biceps",
+              secondary_body_part: "Triceps",
+            },
+          ],
+        };
```
Comment (chelseagreen) [inline ui/__tests__/exercise.test.js:23]: do like this everywhere

File: ui/__tests__/exercise.test.js [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+  describe("Sorting functionality", () => {
+    test("should sort by name in ascending order", async () => {
+      wrapper.vm.sortBy("name");
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises[0].name).toBe("Arm Day");
+    });
+
+    test("should sort by name in descending order", async () => {
+      wrapper.vm.sortBy("name");
+      wrapper.vm.sortBy("name");
```
Comment (chelseagreen) [inline ui/__tests__/exercise.test.js:37]: do this in clients and client workouts

File: ui/__tests__/exercise.test.js [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+            {
+              name: "Leg Day",
+              primary_body_part: "Triceps",
+              secondary_body_part: "Calves",
+            },
+            {
+              name: "Arm Day",
+              primary_body_part: "Biceps",
+              secondary_body_part: "Triceps",
+            },
```
Comment (chelseagreen) [inline ui/__tests__/exercise.test.js:21]: make the test data make sense

File: ui/__tests__/exercise.test.js [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises.length).toBe(2);
+      expect(wrapper.vm.filteredExercises[0].secondary_body_part).toBe(
+        "Calves"
+      );
+      expect(wrapper.vm.filteredExercises[1].secondary_body_part).toBe(
+        "Triceps"
+      );
+    });
+  });
```
Comment (chelseagreen) [inline ui/__tests__/exercise.test.js:110]: this should filter by all fields - update test name

File: ui/src/component/ClientGoal.vue [hunk 16]
```diff
@@ -94,12 +82,13 @@ export default {
...
+
         this.clientFirstName = first_name;
         this.clientLastName = last_name;
         this.clientBirthDay = newBDate(birth_day);
-        this.healthMedsNote = health_note;
-        this.goalsNote = goal_note;
-        this.miscNote = misc_note;
+        this.notes.healthMedsNote = health_note;
+        this.notes.goalsNote = goal_note;
+        this.notes.miscNote = misc_note;
```
Comment (chelseagreen) [inline ui/src/component/ClientGoal.vue:91]: wirte a test to make sure that notes and goals are saved correctly

**Review by chelseagreen — COMMENTED** [review]
make them all like exercises tests and all suites only use one mock data object

File: ui/__tests__/exercise.test.js (hunk 15 above)
Comment (zajonesck) [reply ui/__tests__/exercise.test.js:110]: done __ "should filter exercises across all fields"

File: ui/__tests__/clients.test.js (hunk 8 above)
Comment (zajonesck) [reply ui/__tests__/clients.test.js:14]: done added jimmy green

File: ui/__tests__/clients.test.js (hunk 11 above)
Comment (zajonesck) [reply ui/__tests__/clients.test.js:125]: done

File: README.md [hunk 17]
```diff
@@ -81,34 +88,51 @@ We recommend using [VSCode](https://code.visualstudio.com/) with the following e
 
 ## Deployments
 
-Frontend: After running `npm run build`, push to your `main` branch to deploy the app. Heroku uses the contents of the `dist` folder for deployment.
-Backend: Pushing to the `main` branch will automatically trigger a new Heroku build.
+Frontend: After running `npm run build`, push to your `main` branch to deploy the app.
```
Comment (chelseagreen) [inline README.md:91]: You don't need to run `npm run build` to deploy. Amplify takes care of this for you.

File: README.md [hunk 18]
```diff
@@ -81,34 +88,51 @@ We recommend using [VSCode](https://code.visualstudio.com/) with the following e
...
+Backend: `Amplify push` will automatically trigger a new Amplify build.
 
-### View Heroku logs
+### View Amplify logs
```
Comment (chelseagreen) [inline README.md:94]: I don't think this is the right name for this section

File: README.md [hunk 19]
```diff
@@ -81,34 +88,51 @@ We recommend using [VSCode](https://code.visualstudio.com/) with the following e
...
-With Supertest, we can easily make HTTP requests to our API and assert the responses, making it suitable for testing RESTful services.
+Testing is crucial for ensuring the reliability and stability of the application. In this project, automated tests have been set up to make the validation process more efficient. Ensure all dependencies are installed and the application's environment is set up correctly for the tests to run without issues.
 
-```
-##RUNNING THE TESTS##
+### Tools Used for API Testing
+
+- **Jest**: A delightful JavaScript Testing Framework with a focus on simplicity.
+- **Supertest**: Provides a high-level abstraction for testing HTTP, while still allowing you to drop down to the lower-level API provided by superagent.
+- **vitest**: Vitest is a unit-test framework tailored for Vite, utilizing Vite's dev server for file transformations and offering rapid testing with features like Hot Module Reload (HMR). With a Jest-compatible API, it emphasizes performance and developer experience, aiming to be the preferred test runner for Vite projects.
```
Comment (chelseagreen) [inline README.md:128]: vitest is only for ui testing. Let's separate theise out. Also make sure to specify the commands for how to test

File: README.md [hunk 20]
```diff
@@ -2,74 +2,81 @@
...
 - **Database**: PostgreSQL, running on `localhost:5431`
 
-## Production Overview on Heroku
+## Production Overview on AWS
 
-- **UI**: Vue.js, accessible at [bettertogether.day](https://www.bettertogether.day/#/client-roster)
-- **API**: Node.js, hosted by Heroku and available at `https://better-together-f87fbab820d6.herokuapp.com/`
-- **Database**: PostgreSQL, hosted by Heroku and running on port `5432`
+- **UI**: Vue.js, accessible at [bettertogether.day](https://www.bettertogether.day/)
+- **API**: Node.js, hosted by AWS Lambda functions accessed through API Gateway and available at `https://nn43xtyoif.execute-api.us-east-1.amazonaws.com`
```
Comment (chelseagreen) [inline README.md:25]: Both are deployed using AWS Amplify and CI/CD

File: ui/__tests__/clientWorkout.test.js (hunk 7 above)
Comment (chelseagreen) [reply ui/__tests__/clientWorkout.test.js:125]: @zajonesck

**Review by chelseagreen — APPROVED** [review]

**Review by chelseagreen — APPROVED** [review]

**Pull request merged**